test/acceptance/features/bindToService.feature

```diff
@@ -217,6 +217,42 @@ Feature: Bind an application to a service
         And jq ".status.conditions[] | select(.type=="InjectionReady").status" of Service Binding Request "binding-request-empty-app" should be changed to "False"
         And jq ".status.conditions[] | select(.type=="InjectionReady").reason" of Service Binding Request "binding-request-empty-app" should be changed to "EmptyApplicationSelector"
 
+    Scenario: Bind an imported quarkus app which is deployed as knative service to PostgreSQL database
+        Given Openshift Serverless Operator is running
+        * Quarkus s2i builder image is present
+        * Knative serving is running
+        * DB "db-demo-knative" is running
+        * Quarkus application "knative-app" is imported as Knative service
+        When Service Binding Request is applied to connect the database and the application
+            """
+            ---
+            apiVersion: apps.openshift.io/v1alpha1
+            kind: ServiceBindingRequest
+            metadata:
+              name: binding-request-knative
+            spec:
+              applicationSelector:
+                group: serving.knative.dev
+                version: v1beta1
+                resource: services
+                resourceRef: knative-app
+              backingServiceSelector:
+                group: postgresql.baiju.dev
+                version: v1alpha1
+                kind: Database
+                resourceRef: db-demo-knative
+                id: knav
+              customEnvVar:
+                - name: JDBC_URL
+                  value: jdbc:postgresql://{{ .knav.status.dbConnectionIP }}:{{ .knav.status.dbConnectionPort }}/{{ .knav.status.dbName }}
+                - name: DB_USER
+                  value: "{{ .knav.status.dbCredentials.user }}"
+                - name: DB_PASSWORD
+                  value: "{{ .knav.status.dbCredentials.password }}"
+            """
+        Then application should be connected to the DB "db-demo-knative"
+        And jq ".status.conditions[] | select(.type=="CollectionReady").status" of Service Binding Request "binding-request-knative" should be changed to "True"
+        And jq ".status.conditions[] | select(.type=="InjectionReady").status" of Service Binding Request "binding-request-knative" should be changed to "True"
```

Comment:
Could it be useful to verify injected `secretRef` in `knative service`? Just to add an extra assertion.

Reply:
Need to know more on this. First thing, I believe validating injected `secretRef` is applicable for other scenarios for binding as well. like nodejs. if applicable, then i would like to do that on another PR, else if it is only applicable for binding knative scenario, let me know. Thanks

Reply:
It is applicable for all the scenarios. You can refer that 2 sbr's binding scenario for the same. Just you need to find JSON path for the knative-service scenario.

Reply:
I would handle this in another PR as we need to have a common function for handling number of binding request across the tests (including nodejs) maybe like this: 
`Then "nodejs-app" deployment must contain SBR name "binding-request-1" and "binding-request-2" Examples: | binding_request_name | | binding-request-1 | | binding-request-2 |`

Reply:
We already have a function to fetch `secretRef` but it is for deployment https://github.com/redhat-developer/service-binding-operator/blob/master/test/acceptance/features/steps/openshift.py#L167. 

For Knative service, you'll need to write another one and it will not make sense to raise another PR for it, this function will be very specific for knative-service only so you won't be able to use it for other scenarios.

Reply:
I will look into it for secretRef for knative. meanwhile what do you suggest for nodejs example?

Reply:
We can use this https://github.com/redhat-developer/service-binding-operator/blob/master/test/acceptance/features/steps/openshift.py#L167 for nodejs example.

Reply:
Sure, added this block 
`@ then(u'"{app_name}" deployment must contain SBR name "{sbr_name}"')
def then_envFrom_contains_SBR_name(context, app_name, sbr_name):
    time.sleep(60)
    openshift = Openshift()
    result = openshift.get_deployment_envFrom_info(app_name, context.namespace.name)
    assert re.search(r'.*[map[secretRef:map[name:%s]]' % app_name, result) is not None, f'{app_name} deployment does not contain secretRef: {sbr_name}'
`

Reply:
Could you please check the JSON path for injected secretRef in knative service. If the JSON path is same is deployment then `openshift.get_deployment_envFrom_info()` is useful or you can use a robust method to retrieve elements using JSON path: https://github.com/redhat-developer/service-binding-operator/blob/master/test/acceptance/features/steps/openshift.py#L173

Reply:
it is same for knative as well,
`[shobith@localhost util]$ oc get deploy knative-app-44k86-deployment -n test-namespace-c5501dbe -o "jsonpath={.spec.template.spec.containers[0].envFrom}"
[map[secretRef:map[name:binding-request-knative]]]`
Hence using the same jsonpath path for secretRef